lms/djangoapps/student_account/views.py

```diff
@@ -74,6 +74,24 @@ def login_and_registration_form(request, initial_mode="login"):
     if request.user.is_authenticated():
         return redirect(redirect_to)
 
+    running_pipeline = pipeline.get(request)
+    if running_pipeline and initial_mode == 'register':
+        current_provider = third_party_auth.provider.Registry.get_from_pipeline(running_pipeline)
+        enterprise_customer = enterprise_customer_for_request(request)
+        if enterprise_customer and current_provider.sync_learner_profile_data:
```

Comment:
nit gurad against empty provider:
`if enterprise_customer and current_provider and current_provider.sync_learner_profile_data:`